feat: Add per net device metrics (#4145)
firecracker-microvm/firecracker: CHANGELOG.md

### Added

- [#4145](https://github.com/firecracker-microvm/firecracker/pull/4145):
  Added support for per net device metrics. Now, along with aggregated
  network device metrics, metrics per network device with also be emitted.
  Aggregate metrics will continue to be emitted with label "net" in the
  metrics json object while each net device metrics will be emitted with
  label "net_{iface_id}". E.g. the associated metrics for the endpoint
  "/network-interfaces/eth0" will be available under "net_eth0:" in each

>   metrics json object.

Reviewer: This is out of scope here, but do we plan to do the same for the other devices? It'd be a bit awkward to have a release that has per-device metrics but only for a some of the devices, imo.

> - [#4145](https://github.com/firecracker-microvm/firecracker/pull/4145):
>   Added support for per net device metrics. Now, along with aggregated
>   network device metrics, metrics per network device with also be emitted.
>   Aggregate metrics will continue to be emitted with label "net" in the
>   metrics json object while each net device metrics will be emitted with
>   label "net_{iface_id}". E.g. the associated metrics for the endpoint
>   "/network-interfaces/eth0" will be available under "net_eth0:" in each
>   metrics json object.

Reviewer: nit: This feels a bit verbose to me, I think we can cut out a sentence or two in the middle, e.g. something like

```suggestion
- [#4145](https://github.com/firecracker-microvm/firecracker/pull/4145):
  Added support for per net device metrics. In addition to aggregate metrics,
  each individual net device will emit metrics under the label "net_{iface_id}". 
  E.g. the associated metrics for the endpoint "/network-interfaces/eth0" will
  be available under "net_eth0" in the metrics json object.
```

  capability checks that Firecracker performs during boot. If any of
  these fields are in use, minimal target snapshot version is
  restricted to 1.5.
- [#4145](https://github.com/firecracker-microvm/firecracker/pull/4145):

>   Added support for per net device metrics. Now, along with aggregated

Reviewer 2: nit: "net" and "network" are both used. Could we unify?

Author: this was updated in the later commits.

>   network device metrics, metrics per network device with also be emitted.

Reviewer 2: s/with/will/ ?

Author: fixed in later commits.
